lib/system/chcks.nim

```diff
@@ -32,11 +32,11 @@ proc raiseFieldError(f: string) {.compilerproc, noinline.} =
 when defined(gcdestructors):
   proc raiseFieldError2(f: string, discVal: int) {.compilerproc, noinline.} =
     ## raised when field is inaccessible given runtime value of discriminant
-    sysFatal(FieldError, f & $discVal & "'")
+    sysFatal(FieldDefect, f & $discVal & "'")
```

Comment:
Keep these as errors for now.

Reply:
Any reasons why?

Reply:
Because defects is half-baked and even when it seems irrecoverable, turns out it is recoverable in a number of circumstances. Runtime errors suck, wise yet when we make them a chore to handle.

Reply:
Yea, I wasn't really involved in the discussion, I thought it was just a rename. I will revert the change, but the warning is still there to use Defect instead of Error. So I don't know what to do about it.